src/schema/resolvers.js

```diff
@@ -250,124 +273,245 @@ module.exports = {
         resultSettingStartTime,
         resultSettingEndTime,
       };
+      await DBHelper.insertOracle(Oracles, oracle);
 
+      // Insert Transaction
       const tx = {
         _id: txid,
         txid,
         version,
         type: 'CREATEEVENT',
-        txStatus: 'PENDING',
+        status: 'PENDING',
         senderAddress,
         createdTime: Date.now().toString(),
       };
+      await DBHelper.insertTransaction(Transactions, tx);
+
+      return tx;
+    },
+
+    createBet: async (root, data, { db: { Transactions } }) => {
+      const {
+        version,
+        oracleAddress,
+        optionIdx,
+        amount,
+        senderAddress,
+      } = data;
 
+      // Send bet tx
+      let txid;
       try {
-        await Topics.insert(topic);
-        await Oracles.insert(oracle);
-        await Transactions.insert(tx);
+        const tx = await centralizedOracle.bet({
+          contractAddress: oracleAddress,
+          index: optionIdx,
+          amount,
+          senderAddress,
+        });
+        txid = tx.txid;
       } catch (err) {
-        logger.error(`Error insertion db: ${err.message}`);
+        logger.error(`Error calling CentralizedOracle.bet: ${err.message}`);
         throw err;
       }
 
+      // Insert Transaction
+      const tx = {
+        _id: txid,
+        txid,
+        version,
+        type: 'BET',
+        status: 'PENDING',
+        senderAddress,
+        oracleAddress,
+        optionIdx,
+        token: 'QTUM',
+        amount,
```

Comment:
convert amount from `decimalstring` to `hexstring`

Reply:
same note as above